piptools/scripts/compile.py

```diff
@@ -444,6 +444,15 @@ def cli(
     # Proxy with a LocalRequirementsRepository if --upgrade is not specified
     # (= default invocation)
     if not upgrade and os.path.exists(output_file.name):
+        if upgrade_install_reqs and os.path.getsize(output_file.name) == 0:
```

Comment:
I noticed that this conditional clause is getting complicated, which affects readability because of the perceived cognitive complexity.
How about labeling the conditions with nice variable names?
```suggestion
    output_file_exists = os.path.exists(output_file.name)
    output_file_is_empty = True if not output_file_exists else os.path.getsize(output_file.name) == 0
    if not upgrade and output_file_exists:
        if upgrade_install_reqs and output_file_is_empty:
```

Reply:
```python
output_file_is_empty = not output_file_exists or os.path.getsize(output_file.name) == 0
```
I personally feel it's better like this:

```python
output_file_is_empty = os.path.getsize(output_file.name) == 0
```

Since it is already checked in the prior if condition if the file exists. But that is just a small improvement to code readability.

Reply:
I opted for @chysle's formulation (although open to suggestion!) - I felt that moving both Booleans out of the conditional was nice, but required an `or` that was itself tricky to read.

Reply:
Fair enough. Marking the thread as resolved.